### PR TITLE
Remove shared informer factory from KCM

### LINKED ISSUE #3201: Remove shared informer factory from kubermatic controller manager
**User Story**

As a developer of Kubermatic I want it to contain as little code as possible. As a user of Kubermatic, I want it to use as little memory as possible. Both can be improved by removing the uneeded shared informer factory from the kubermatic cluster controller

**Acceptance criteria**

* The kubermatic cluster controller only creates the caches managed by controller-runtime

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3201 

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/cmd/kubermatic-controller-manager/main.go [hunk 1]
```diff
@@ -79,12 +69,10 @@ func main() {
 		glog.Fatalf("failed to add kubermatic scheme to mgr: %v", err)
 	}
 
-	dynamicClient := mgr.GetClient()
-	dynamicCache := mgr.GetCache()
 	recorder := mgr.GetRecorder(controllerName)
 
-	// Check if the CRD for the VerticalPodAutoscaler is registered by allocating an informer
-	if _, err := informer.GetSyncedStoreFromDynamicFactory(dynamicCache, &autoscalingv1beta2.VerticalPodAutoscaler{}); err != nil {
+	// Check if the CRD for the VerticalPodAutoscaler is registered by doing allocating an informer
```
Comment (mrIncompetent) [inline api/cmd/kubermatic-controller-manager/main.go]: > by doing allocating an informer

Grammar looks weird, is this correct?

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated the comment, ptal

**Pull request merged**